seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-assert/src/main/java/org/apache/seatunnel/flink/assertion/sink/AssertSink.java

```diff
@@ -59,7 +61,7 @@ public void outputBatch(FlinkEnvironment env, DataSet<Row> inDataSet) {
                     throw new IllegalStateException("row :" + row + " fail rule: " + failRule);
                 });
             return null;
-        });
+        }).print();
```

Comment:
It's better to use collect here, if you don't want to print each row.
```
try {
            inDataSet.collect().forEach(row -> {
                ASSERT_EXECUTOR
                    .fail(row, assertFieldRules)
                    .ifPresent(failRule -> {
                        throw new IllegalStateException("row :" + row + " fail rule: " + failRule);
                    });
            });
        } catch (IllegalArgumentException illegalArgumentException) {
            throw illegalArgumentException;
        } catch (Exception ex) {
            throw new RuntimeException("AssertSink execute failed", ex);
        }
```

Reply:
This will cause:
```
The program finished with the following exception:

org.apache.flink.client.program.ProgramInvocationException: The main method caused an error: Execute Flink task error
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:372)
	at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:222)
	at org.apache.flink.client.ClientUtils.executeProgram(ClientUtils.java:114)
	at org.apache.flink.client.cli.CliFrontend.executeProgram(CliFrontend.java:812)
	at org.apache.flink.client.cli.CliFrontend.run(CliFrontend.java:246)
	at org.apache.flink.client.cli.CliFrontend.parseAndRun(CliFrontend.java:1054)
	at org.apache.flink.client.cli.CliFrontend.lambda$main$10(CliFrontend.java:1132)
	at org.apache.flink.runtime.security.contexts.NoOpSecurityContext.runSecured(NoOpSecurityContext.java:28)
	at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:1132)
Caused by: java.lang.RuntimeException: Execute Flink task error
	at org.apache.seatunnel.core.flink.command.FlinkTaskExecuteCommand.execute(FlinkTaskExecuteCommand.java:84)
	at org.apache.seatunnel.core.base.Seatunnel.run(Seatunnel.java:39)
	at org.apache.seatunnel.core.flink.SeatunnelFlink.main(SeatunnelFlink.java:33)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:355)
	... 8 more
Caused by: java.lang.RuntimeException: No new data sinks have been defined since the last execution. The last execution refers to the latest call to 'execute()', 'count()', 'collect()', or 'print()'.
	at org.apache.flink.api.java.ExecutionEnvironment.createProgramPlan(ExecutionEnvironment.java:1165)
	at org.apache.flink.api.java.ExecutionEnvironment.getExecutionPlan(ExecutionEnvironment.java:1076)
	at org.apache.seatunnel.flink.batch.FlinkBatchExecution.start(FlinkBatchExecution.java:76)
	at org.apache.seatunnel.core.flink.command.FlinkTaskExecuteCommand.execute(FlinkTaskExecuteCommand.java:81)
	... 15 more

```

Reply:
`ConsoleSink` can pass CI because it was exclusive by `org.apache.seatunnel.flink.batch.FlinkBatchExecution#whetherExecute`

Reply:
I have added AssertSink in whetherExecute

Reply:
OK, I got you, since when we execute collect, the flink job will be executed, so we cannot execute it again.